assets/js/stripe-payment-request.js

```diff
@@ -675,7 +683,11 @@ jQuery( function( $ ) {
 								displayItems: response.displayItems,
 							} )
 						).then( function () {
-							$( document.body ).trigger( 'wc_stripe_unblock_payment_request_button' );
+							if (response.is_in_stock) {
```

Comment:
Not sure waiting for the backend response to show/hide the PRB is best here. I mean, it'd be the best option if we also updated the whole product data in the frontend (which we currently do not). I think we should get the `is_in_stock` for each variant from the form (`$('variations_form cart').data('product_variations')[0].is_in_stock;`) and show/hide the button accordingly.

Reply:
That's true: waiting until the backend responds to update the PRB does seem a bit odd. Will update 👍